barretenberg/cpp/src/barretenberg/vm/avm/tests/range_check.test.cpp

```diff
@@ -0,0 +1,140 @@
+        trace[i].lookup_rng_chk_0_counts = range_check_builder.u16_range_chk_counters[0][uint16_t(i)];
+        trace[i].lookup_rng_chk_1_counts = range_check_builder.u16_range_chk_counters[1][uint16_t(i)];
+        trace[i].lookup_rng_chk_2_counts = range_check_builder.u16_range_chk_counters[2][uint16_t(i)];
+        trace[i].lookup_rng_chk_3_counts = range_check_builder.u16_range_chk_counters[3][uint16_t(i)];
+        trace[i].lookup_rng_chk_4_counts = range_check_builder.u16_range_chk_counters[4][uint16_t(i)];
+        trace[i].lookup_rng_chk_5_counts = range_check_builder.u16_range_chk_counters[5][uint16_t(i)];
+        trace[i].lookup_rng_chk_6_counts = range_check_builder.u16_range_chk_counters[6][uint16_t(i)];
+        trace[i].lookup_rng_chk_7_counts = range_check_builder.u16_range_chk_counters[7][uint16_t(i)];
+        trace[i].lookup_rng_chk_diff_counts = range_check_builder.dyn_diff_counts[uint16_t(i)];
+        trace[i].lookup_rng_chk_pow_2_counts = range_check_builder.powers_of_2_counts[uint16_t(i)];
+    }
+    std::cerr << "Done generating trace..." << std::endl;
+
+    // We build the polynomials needed to run "sumcheck".
+    AvmCircuitBuilder cb;
+    cb.set_trace(std::move(trace));
+    auto polys = cb.compute_polynomials();
+    const size_t num_rows = polys.get_polynomial_size();
+    std::cerr << "Done computing polynomials..." << std::endl;
+
+    std::cerr << "Accumulating relations..." << std::endl;
+    using Relation = Avm_vm::range_check<FF>;
+
+    typename Relation::SumcheckArrayOfValuesOverSubrelations result;
+    for (auto& r : result) {
+        r = 0;
+    }
+
+    // We set the conditions up there.
+    for (size_t r = 0; r < num_rows; ++r) {
+        Relation::accumulate(result, polys.get_row(r), {}, 1);
+    }
+
+    for (size_t j = 0; j < result.size(); ++j) {
+        if (result[j] != 0) {
+            EXPECT_EQ(result[j], 0) << "Relation " << Relation::NAME << " subrelation "
+                                    << Relation::get_subrelation_label(j) << " was expected to be zero.";
+        }
+    }
+    std::cerr << "Accumulating lookup relations..." << std::endl;
+
+    // Let's be explicit about the lookups we are checking
+    using AllLookupRelations = std::tuple<
+        // Lookups
+        lookup_rng_chk_0_relation<FF>,
+        lookup_rng_chk_1_relation<FF>,
+        lookup_rng_chk_2_relation<FF>,
+        lookup_rng_chk_3_relation<FF>,
+        lookup_rng_chk_4_relation<FF>,
+        lookup_rng_chk_5_relation<FF>,
+        lookup_rng_chk_6_relation<FF>,
+        lookup_rng_chk_7_relation<FF>,
+        lookup_rng_chk_pow_2_relation<FF>,
+        lookup_rng_chk_diff_relation<FF>>;
+
+    const FF gamma = FF::random_element();
+    const FF beta = FF::random_element();
+    bb::RelationParameters<typename AvmFlavor::FF> params{
+        .beta = beta,
+        .gamma = gamma,
+    };
+    bb::constexpr_for<0, std::tuple_size_v<AllLookupRelations>, 1>([&]<size_t i>() {
```

Comment:
This is ok for now, at some point I'll just create helpers to do this for us.

Reply:
Definitely if we want proper unit tests per trace we need to code generate checks pertaining to a trace. I prefer not have such code introduced like this.